### PR TITLE
Add constants to packages and cleanup code

### PR DESCRIPTION
**Pull request by vchepeli**
**Describe what this PR does and why we need it**:
We need to improve out QoC, add additional checks not null and throw additional runtime exceptions

**Changes proposed in this pull request**
 - Add Constants to packages
 - Change legacy code
 - Improve JSon output
- Use threadpool executor for all CLIRunner tasks

### CODE REVIEW

File: src/main/java/org/aerogear/plugin/intellij/mobile/api/CLIRunnerImpl.java [hunk 1]
```diff
@@ -12,17 +12,20 @@
...
+
+  private static final int CMD_TIMEOUT_SECONDS = 10;
+  private static final int EXECUTOR_NUM_THREADS = 3;
+
+  private final MobileNotificationsService notificationsService;
+  private final ExecutorService executorService;
+
   public CLIRunnerImpl() {
     this.notificationsService = new MobileNotificationsService();
+    this.executorService = Executors.newFixedThreadPool(EXECUTOR_NUM_THREADS);
```
Comment (sedroche) [inline src/main/java/org/aerogear/plugin/intellij/mobile/api/CLIRunnerImpl.java]: Is it safe to create this per instance or for each call to `executeAsync`? I am unsure of how the Executor classes work

File: src/main/java/org/aerogear/plugin/intellij/mobile/services/sdkconfig/SDKConfigManager.java [hunk 2]
```diff
@@ -12,24 +12,28 @@
...
-            ApplicationManager.getApplication().runReadAction(openConfigRunner);
-            AeroGearMobileConfiguration.getInstance(project).setConfigPath(path);
-        });
+            this.asyncRefreshFiles(() -> {
+                WriteCommandAction.runWriteCommandAction(project, updateConfigRunner);
+                ApplicationManager.getApplication().runReadAction(openConfigRunner);
+                Objects.requireNonNull(AeroGearMobileConfiguration.getInstance(project)).setConfigPath(path);
+            });
+        } else {
+            Objects.requireNonNull(AeroGearMobileConfiguration.getInstance(project)).setConfigPath(path);
```
Comment (sedroche) [inline src/main/java/org/aerogear/plugin/intellij/mobile/services/sdkconfig/SDKConfigManager.java]: If the file is not created why do we want to store the setting?

File: src/main/java/org/aerogear/plugin/intellij/mobile/ui/sdkconfig/NewFileForm.java [hunk 3]
```diff
@@ -3,7 +3,7 @@
 import javax.swing.JTextField;
 import com.intellij.openapi.ui.TextFieldWithBrowseButton;
 
-public class NewFileForm extends javax.swing.JPanel {
```
Comment (sedroche) [inline src/main/java/org/aerogear/plugin/intellij/mobile/ui/sdkconfig/NewFileForm.java]: Any classes that are auto-generated UI classes I would like to leave alone.

File: src/main/java/org/aerogear/plugin/intellij/mobile/ui/sdkconfig/ServiceDeployedNotification.java [hunk 4]
```diff
@@ -31,38 +28,39 @@ public ServiceDeployedNotification(Project project, ServiceClass sc) {
     }
 
     private String createContent() {
-        String sdkConfigPath = AeroGearMobileConfiguration.getInstance(this.project).getConfigPath();
-        String content = "";
+        String sdkConfigPath = Objects.requireNonNull(AeroGearMobileConfiguration.getInstance(this.project)).getConfigPath();
+        StringBuilder content = new StringBuilder(); // TODO ADD template files if notification is redundant
```
Comment (sedroche) [inline src/main/java/org/aerogear/plugin/intellij/mobile/ui/sdkconfig/ServiceDeployedNotification.java]: Is this TODO an issue? There is one created around template file already. Does that cover it?
https://github.com/aerogear/aerogear-mobile-intellij-plugin/issues/19

File: src/main/java/org/aerogear/plugin/intellij/mobile/ui/sdkconfig/ViewSDKConfigPanel.java [hunk 5]
```diff
@@ -2,15 +2,15 @@
...
-public class ViewSDKConfigPanel extends javax.swing.JPanel {
+class ViewSDKConfigPanel extends javax.swing.JPanel {
 
     public ViewSDKConfigPanel() {
         initComponents();
     }
 
     private void initComponents() {
 
-        scrollPane = new javax.swing.JScrollPane();
```
Comment (sedroche) [inline src/main/java/org/aerogear/plugin/intellij/mobile/ui/sdkconfig/ViewSDKConfigPanel.java]: Any classes that are auto-generated UI classes I would like to leave alone.
If it is could you put a comment in like the other auto-generated files?

File: src/main/java/org/aerogear/plugin/intellij/mobile/ui/servicecatalog/ServicePanel.java [hunk 6]
```diff
@@ -14,7 +16,7 @@
  * To edit this file, open the ServicePanel.form file in the NetBeans GUI Builder.
  * To view the generated source code, choose Source within the Netbeans GUI Builder.
  */
-public class ServicePanel extends javax.swing.JPanel {
+class ServicePanel extends javax.swing.JPanel {
```
Comment (sedroche) [inline src/main/java/org/aerogear/plugin/intellij/mobile/ui/servicecatalog/ServicePanel.java]: Any classes that are auto-generated UI classes I would like to leave alone.

File: src/main/java/org/aerogear/plugin/intellij/mobile/ui/servicecatalog/identity/IdentityConfigurationPanel.java [hunk 7]
```diff
@@ -6,15 +6,17 @@
...
-import javax.swing.JTextField;
+import com.intellij.ui.Gray;
+import com.intellij.ui.JBColor;
+
+import javax.swing.*;
 
 /**
  * This class is generated by and can be edited using the Netbeans GUI Builder.
  * To edit this file, open the ServicePanel.form file in the NetBeans GUI Builder.
  * To view the generated source code, choose Source within the Netbeans GUI Builder.
```
Comment (sedroche) [inline src/main/java/org/aerogear/plugin/intellij/mobile/ui/servicecatalog/identity/IdentityConfigurationPanel.java]: same

File: src/main/java/org/aerogear/plugin/intellij/mobile/services/sdkconfig/SDKConfigManager.java (hunk 2 above)
Comment (vchepeli) [reply src/main/java/org/aerogear/plugin/intellij/mobile/services/sdkconfig/SDKConfigManager.java]: File you want to create already exists and you need only update your configPath to be same as path

**Comment by sedroche** [comment]
All good from my point @witmicko @vchepeli

**Pull request merged**